### PR TITLE
Add code formatter `yapf` to the tests and pre-commit

### LINKED ISSUE #482: ci: Introduce code formatter
## 🚀 Feature
We've spent quite some time on fixing code formatting in code reviews by our own hands, but this should be done automatically by a Python code formatter.

### List of formatters
- `black` by Python Software Foundation (https://github.com/psf/black)
  - [This blog post by DeepSource](https://deepsource.io/blog/python-code-formatters/) says `black` is the most popular one.
- `yapf` by Google (https://github.com/google/yapf)
- (`autopep8`, ...)

### Additional context
I have used `black` with `isort` (for import formatting) and `flake8` (for pep8 check) in [Optuna](https://github.com/optuna/optuna), and it worked really well, so +1 for `black`.

In PL, `black` was added in https://github.com/PyTorchLightning/pytorch-lightning/pull/1610, but it hasn't been really used... We can start to employ the code formatter here, and if there seems no problem with it, we can enable it in PL later, too.

cc: @Borda @PyTorchLightning/bolts-contributors

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Adds `yapf` https://github.com/google/yapf to CI and pre-commit hooks and is part of #482.

### TODO
- [x] add `yapf` to pre-commit hooks
- [x] add `yapf` to GitHub Actions
- [x] update `flake8` and `isort` configs to avoid conflicts with `yapf`
  - As far as I looked through some issues and PRs, we don't need to change flake8/isort configs.
- [x] Apply `yapf` to other files than pl_bolts/ and tests/.

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes?
- [x] Did you write any new necessary tests? [not needed for typos/docs]
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)?

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,9 +1,19 @@
+default_language_version:
+  python: python3.8
```
Comment (akihironitta) [inline .pre-commit-config.yaml:2]: Added this to match with the config in PL

**Comment by akihironitta** [comment]
Applied `yapf` in https://github.com/PyTorchLightning/pytorch-lightning-bolts/pull/510, and it looks neat.

File: docs/source/conf.py [hunk 2]
```diff
@@ -304,13 +300,10 @@ def run_apidoc(_):
...
-        argv = ['-e',
-                '-o', apidoc_output_folder,
-                os.path.join(PATH_ROOT, pkg),
-                '**/test_*',
-                '--force',
-                '--private',
-                '--module-first']
+        argv = [
+            '-e', '-o', apidoc_output_folder,
+            os.path.join(PATH_ROOT, pkg), '**/test_*', '--force', '--private', '--module-first'
```
Comment (Borda) [inline docs/source/conf.py:305]:
```suggestion
            os.path.join(PATH_ROOT, pkg), '**/test_*', '--force', '--private', '--module-first',
```

File: docs/source/conf.py [hunk 3]
```diff
@@ -231,19 +227,19 @@
...
 #  dir menu entry, description, category)
 texinfo_documents = [
-    (master_doc, project, project + ' Documentation', author, project,
-     'The lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerplate.',
-     'Miscellaneous'),
+    (
+        master_doc, project, project + ' Documentation', author, project,
+        'The lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerplate.',
+        'Miscellaneous'
+    ),
```
Comment (Borda) [inline docs/source/conf.py:242]:
```suggestion
    )
```

File: docs/source/conf.py [hunk 4]
```diff
@@ -231,19 +227,19 @@
...
 # (source start file, target name, title, author,
 #  dir menu entry, description, category)
 texinfo_documents = [
-    (master_doc, project, project + ' Documentation', author, project,
-     'The lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerplate.',
-     'Miscellaneous'),
+    (
+        master_doc, project, project + ' Documentation', author, project,
+        'The lightweight PyTorch wrapper for ML researchers. Scale your models. Write less boilerplate.',
+        'Miscellaneous'
```
Comment (Borda) [inline docs/source/conf.py:241]:
```suggestion
        'Miscellaneous',
```

**Review by Borda — APPROVED** [review]

**Pull request merged**